src/llvm-cpufeatures.cpp

```diff
@@ -62,15 +62,18 @@ static bool have_fma(Function &intr, Function &caller, const Triple &TT) JL_NOTS
     SmallVector<StringRef, 128> Features;
     FS.split(Features, ',');
     for (StringRef Feature : Features)
-    if (TT.isARM()) {
-      if (Feature == "+vfp4")
-        return typ == "f32" || typ == "f64";
-      else if (Feature == "+vfp4sp")
-        return typ == "f32";
-    } else if (TT.isX86()) {
-      if (Feature == "+fma" || Feature == "+fma4")
-        return typ == "f32" || typ == "f64";
-    }
+        if (TT.isARM()) {
+            if (Feature == "+vfp4")
+                return typ == "f32" || typ == "f64";
+            else if (Feature == "+vfp4sp")
+                return typ == "f32";
+        } else if (TT.isX86()) {
+            if (Feature == "+fma" || Feature == "+fma4")
+                return typ == "f32" || typ == "f64";
+        } else if (TT.isRISCV64()) {
+            if (Feature == "+zfh" || Feature == "+f" || Feature == "+d")
+                return typ == "f16" || typ == "f32" || typ == "f64";
+        }
```

Comment:
I think this implication is to broad. Proably should be `Feature == "+zfh"`  -> `typ == "f16"`

Reply:
https://github.com/riscv/riscv-isa-manual/blob/main/src/d-st-ext.adoc

So D implies F, like on ARM64.

https://github.com/riscv/riscv-isa-manual/blob/main/src/zfh.adoc

ZFH implies F